joomla/joomla-cms
# Fix actionlogs information emails containing HTML links

Pull Request for Issue #39851. This is essentially my strategy "3" as outlined in the issue, with the difference that the two replacement arrays are not merged. As I did not receive any comments on the issue, I just went ahead to implement this solution....

### Summary of Changes

This needed multiple changes:
1. `MailTemplate` now supports an extra set of replacement values for the plain-text mode. This will only be used if set explicitely, and will fall back to the old behaviour otherwise. The 'default' data and the plain data are _not_ merged. Callers have to ensure that both sets of values contain the same keys, or otherwise output might be missing.
1. `ActionlogModel` uses this new facility to provide values without HTML links.
1. `ActionlogsHelper::getHumanReadableLogMessage()` already had a parameter `$generateLinks`, that did not do what was documented. It only controlled the conversion from relative to absolute URLs. Now it will also remove link tags from the message templates, if set to false.

**Note**: This also affects the CSV export feature of the actionlogs components. Until now, the exported files contained relative urls in the administrator application (which is - I think - quite useless anyway). With this PR, links are also stripped. (This is the the only other call site that changed the value of `$generateLinks` to false.)


### Testing Instructions
* Install a recent 4.3.0 nighly build, that already has #39850 upmerged.
* Make sure action logs for user events are enabled and configure working system email sending
* In the user settings for a user with a valid email address (you can receive mail on), enable user action information emails for the user component.
* Log out and log back in
* Look at the emails you should have received


### Actual result BEFORE applying this Pull Request
Plain text emails contain HTML links


### Expected result AFTER applying this Pull Request
Plain text emails do not contain HTML links and only the link text.


### Link to documentations
Please select:
